server/svix-server/src/queue/redis.rs

```diff
@@ -234,3 +215,123 @@ impl TaskQueueReceive for RedisQueueConsumer {
         Ok(from_redis_key(&key))
     }
 }
+
+async fn migrate_legacy_queues(pool: &mut PooledConnection<'_>) {
+    migrate_list(pool, LEGACY_MAIN, MAIN).await;
+    migrate_list(pool, LEGACY_PROCESSING, PROCESSING).await;
+    migrate_sset(pool, LEGACY_DELAYED, DELAYED).await;
+}
+
+async fn migrate_list(pool: &mut PooledConnection<'_>, legacy_queue: &str, queue: &str) {
+    let batch_size = 1000;
+    let mut x = 0;
+    loop {
```

Comment:
It feels confusing to only wipe the queue once you migrated all of them, instead of after processing each batch. Also the way the condition for the wiping works is very implicit (oh, the last range fetch returned nothing it means we are past the end). I liked how you had it before better.